packages/wallets/src/features/cfd/flows/Verification/Verification.tsx

```diff
@@ -210,14 +210,23 @@ const Verification: FC<TVerificationProps> = ({ selectedJurisdiction }) => {
             } else if (currentScreenId === 'manualScreen') {
                 switchScreen('selfieScreen');
             } else if (currentScreenId === 'poaScreen') {
+                await upload({
+                    document_issuing_country: settings?.country_code ?? undefined,
+                    document_type: 'proofaddress',
+                    file: formValues.poaDocument,
+                });
                 updateSettings({
                     address_city: formValues.townCityLine,
                     address_line_1: formValues.firstLine,
                     address_line_2: formValues.secondLine,
                     address_postcode: formValues.zipCodeLine,
                     address_state: formValues.stateProvinceDropdownLine,
                 });
-                switchScreen('personalDetailsScreen');
+                if (!settings?.has_submitted_personal_details) {
```

Comment:
Sometimes this would not work. I think its better to put it in a variable to store it.